src/values/AbstractValue.js

```diff
@@ -278,6 +296,28 @@ export default class AbstractValue extends Value {
     return result;
   }
 
+  refineWithPathCondition(): Value {
+    if (this.kind !== "conditional") return this;
+    let [condition, trueVal, falseVal] = this.args;
+    if (!(condition instanceof AbstractValue && condition.getType() === BooleanValue)) return this;
```

Comment:
How can the condition not be an abstract value?
Even if the condition can be of a non-BooleanValue type, (why) does it matter?

Reply:
Right. This should be an invariant and the type does not matter.